client/me/notification-settings/blogs-settings/header.jsx

```diff
@@ -45,29 +44,31 @@ class BlogSettingsHeader extends PureComponent {
 	};
 
 	getLegend = () => {
-		const tally = o => o.reduce( ( total, value ) => total + value );
-		const sizeAndSum = settings => [ settings.size, tally( settings ) ];
-
-		let counts = this.props.settings
-			.deleteIn( [ 'email', 'achievement' ] )
-			.filterNot( ( _, key ) => includes( [ 'blog_id', 'devices' ], key ) )
-			.map( sizeAndSum )
-			.toArray();
-
-		counts = this.props.settings
-			.get( 'devices' )
-			.map( device => device.filter( ( _, key ) => key !== 'device_id' ) )
-			.map( sizeAndSum )
-			.toArray()
-			.concat( counts );
-
-		const [ size, count ] = zip.apply( null, counts ).map( tally );
+		const { settings } = this.props;
+
+		// Ignore blog_id, email.achievement and devices (we'll handle devices separately).
+		const filteredSettings = {
+			...omit( settings, [ 'blog_id', 'devices' ] ),
+			email: omit( get( settings, 'email', {} ), 'achievement' ),
+		};
+		// Ignore the device_id of each device found.
+		const devicesSettings = map( settings.devices, device => omit( device, 'device_id' ) );
+
+		const [ onSettings, offSettings ] = partition(
+			flatten( [ ...map( filteredSettings, values ), ...map( devicesSettings, values ) ] )
+		);
```

Comment:
`_.countBy` could be useful: the result is `[ onSettingsCount, offSettingsCount ]` instead of two arrays. Better for our purpose here.